lib/elixir/lib/kernel/parallel_compiler.ex

```diff
@@ -176,12 +178,12 @@ defmodule Kernel.ParallelCompiler do
     end
   end
 
-  defp spawn_workers(schedulers, checker, files, output, options) do
+  defp spawn_workers(schedulers, checker, files, output, compilation_root, options) do
     threshold = Keyword.get(options, :long_compilation_threshold, 10) * 1000
     timer_ref = Process.send_after(self(), :threshold_check, threshold)
 
     {outcome, state} =
-      spawn_workers(files, 0, [], [], %{}, [], %{
+      spawn_workers(files, 0, [], [], %{}, [], compilation_root, %{
```

Comment:
Let's add the compilation root to the state map below instead. That will require less changes overall.

Reply:
I'm happy to do that. My initial thought was that since it's supposed to be more or less a fixed constant passed from the top, an entry in a state map may imply subjectivity to change, but I am more than happy to change it if you think it's more appropriate.

Reply:
Hmm, on second inspection, I could add it to the state map below, but only really for the spawn_workers/7 handle, since the others would only have the `opts` parameter, and it doesn't feel appropriate to pass a constant as an opt. So the other spawn_workers would stay the same. Does this sound okay?